### PR TITLE
Retry if request failed due to rate-limit

### PR DESCRIPTION
**Pull request by anewusername**
Temporary fix for #6 so things stop failing due to rate-limits.
If the request fails, waits retry_after_ms and then resends it.

Signed-off-by: Jan Petykiewicz <anewusername@gmail.com>

### CODE REVIEW

**Comment by Half-Shot** [comment]
ok to test

**Comment by Half-Shot** [comment]
LGTM

File: matrix_client/api.py [hunk 1]
```diff
@@ -434,13 +434,20 @@ def _send(self, method, path, content=None, query_params={}, headers={},
...
 
-        response = requests.request(
-            method, endpoint,
-            params=query_params,
-            data=content,
-            headers=headers,
-            verify=self.validate_cert
-        )
+        response = None
+        while True:
```
Comment (pik) [inline matrix_client/api.py]: Adding blocking code the default `_send` method in the SDK is really not a nice idea. Clients using the SDK can have different concurrency models and dealing with re-tries is much better left to them.

Comment (Half-Shot) [inline matrix_client/api.py]: Hmm, you might be right on this one. It probably would make more sense to make this optional.

**Pull request merged**